### DISCUSSION TITLE
Open directly an aedt file

### DISCUSSION THREAD

**Discussion by Herve-L1** [ansys/pyaedt#4680] [Q&A] [answered]
How to open directly an aedt file?

In Maxwell Scripting I do this:

import ScriptEnv
# aedt_ver = 'Ansoft.ElectronicsDesktop'
# oAnsoftApp = win32com.client.Dispatch(aedt_ver)
# oDesktop = oAnsoftApp.GetAppDesktop()

#ScriptEnv.Initialize("Ansoft.ElectronicsDesktop")
ScriptEnv.InitializeNew(NonGraphical=True)

oDesktop.OpenProject(work_file)
oDesktop.RestoreWindow()

Thanks in advance for your help.

**Comment by Samuelopez-ansys** [comment, marked as answer]
@Herve-L1 The script you provided is how you can interact directly with the AEDT API. But PyAEDT provides an extra layer to avoid this call, or at least to make it not visible.

You can:

from pyaedt import Desktop

# Open AEDT
desktop= Desktop()

# Load project
aedtapp = desktop.load_project(work_file)

# Your automation

# Release AEDT
desktop.release_desktop()

This is just one option, if you know the application to open, for example HFSS, you can do 

from pyaedt import Hfss

app = Hfss(projectname=work_file)

Here you can find more examples:
https://aedt.docs.pyansys.com/version/stable/User_guide/index.html

**Comment by Herve-L1** [comment]
Hi Samuel

My problem is solved, sorry for this late reply.

Best regards

Hervé

De : Samuel Lopez ***@***.***>
Envoyé : vendredi 17 mai 2024 15:24
À : ansys/pyaedt ***@***.***>
Cc : Lavalee, Herve (GE Vernova) ***@***.***>; Mention ***@***.***>
Objet : EXT: Re: [ansys/pyaedt] Open directly an aedt file (Discussion #4680)

AVERTISSEMENT: cet email provient de l'extérieur de GE. Veuillez valider l'adresse e-mail de l'expéditeur avant de cliquer sur les liens ou les pièces jointes, car ils risquent de ne pas être sûrs.

Closed #4680<https://github.com/ansys/pyaedt/discussions/4680> as resolved.
